labs/sysdev-buildroot-stm32mp2/sysdev-buildroot-stm32mp2.tex

```diff
@@ -0,0 +1,577 @@
+  Buildroot: everything works through Makefiles in Buildroot;
+\item \code{package} is a directory that contains all the Makefiles,
+  patches and configuration items to compile the user space
+  applications and libraries of your embedded Linux system. Have a
+  look at various subdirectories and see what they contain;
+\item \code{system} contains the root filesystem skeleton and the {\em
+    device tables} used when a static \code{/dev} is used;
+\item \code{toolchain} contains the Makefiles, patches and
+  configuration items to generate the cross-compiling toolchain.
+\end{itemize}
+
+\section{Board specific configuration}
+
+As we will want Buildroot to build a kernel with a custom configuration,
+and our custom patch, so let's add our own subdirectory under
+\code{board}:
+
+\begin{bashinput}
+mkdir -p board/bootlin/training
+\end{bashinput}
+
+Then, copy your kernel configuration and kernel patch:
+
+\begin{bashinput}
+cp ../../kernel/linux/.config board/bootlin/training/linux.config
+cp ../../kernel/linux/0001-Custom-DTS-for-Bootlin-lab.patch \
+   board/bootlin/training/
+\end{bashinput}
+
+We now need to download additional files so that Buildroot can support the board.
+To do this, we will fetch the sources provided by Bootlin.
+
+\begin{bashinput}
+$ cd ..
+$ git clone https://github.com/bootlin/buildroot-external-st.git
+\end{bashinput}
+
+We will configure Buildroot to use this kernel configuration.
+
+\section{Configure Buildroot}
+
+In our case, we would like to:
+
+\begin{itemize}
+\item Generate an embedded Linux system for ARM;
+\item Use an already existing external toolchain instead of having
+  Buildroot generating one for us;
+\item Compile the Linux kernel and deploy its modules in the root
+  filesystem;
+\item Integrate {\em BusyBox}, {\em alsa-utils},
+  {\em mpd}, {\em mpc} and {\em evtest} in our embedded Linux system;
+\item Integrate the target filesystem into a tarball
+\end{itemize}
+
+To run the configuration utility of Buildroot, simply run:
+
+\bashcmd{$ make menuconfig}
+
+Set the following options. Don't hesitate to press the \code{Help}
+button whenever you need more details about a given option:
+
+\begin{itemize}
+\item \code{Target options}
+  \begin{itemize}
+  \item \code{Target Architecture}: \code{AArch64 (little endian)}
+  \item \code{Target Architecture Variant}: \code{cortex-A53}
+  \item \code{Target ABI}: \code{EABIhf}
+  \item \code{Floating point strategy}: \code{FP-ARMv8}
+  \end{itemize}
+\item \code{Toolchain}
+  \begin{itemize}
+  \item \code{Toolchain type}: \code{External toolchain}
+  \item \code{Toolchain}: \code{Custom toolchain}
+  \item \code{Toolchain path}: use the toolchain you built:
+  \code{/home/<user>/x-tools/aarch64-training-linux-musl}
+    (replace \code{<user>} by your actual user name)
+  \item \code{External toolchain gcc version}: \code{13.x}
+  \item \code{External toolchain kernel headers series}: \code{6.13}
```

Comment:
Using a toolchain that has been built with 6.13 headers and then run a 6.6 Linux kernel is incorrect, don't do that. Build a toolchain that uses 6.6.x kernel headers from the start.